### DISCUSSION TITLE
Conditional show/hide table column and conditional row color

### DISCUSSION THREAD

**Discussion by magickaito** [Budibase/budibase#6261] [Help] [answered]
Is it possible to:
1) show/hide a table column based on user role? (eg. only show a column for POWER user but hide it for BASIC user)

It appears that I can only show/hide a table container (on left) based on roles, but can't do this on a column?

2) put background color of a row to red only if a condition is met? 

It appears that I can set a static column color, but not a conditional row color?

I found improvements #3705 mentioned that "All of the suggestions in the discussion #1913 will be addressed". And #1913 mentioned **Colouring rows and cells** and **conditional show/hide of columns**. But I couldn't figure out how to do above two actions from [Conditional UI documentation](https://docs.budibase.com/docs/conditions).

**Comment by dannyshin76** [comment]
Built-in tables doesn't support those features yet. However, you can build your own table component (by using a data provider and repeater) to implement everything you need. It's quite a bit of work regarding formatting and everything, but definitely possible.

**Comment by saz33m1** [comment]
@dannyshin76 Its been over 2 years. Is the following still not possible? Thanks in advance 🙏🏽

`show/hide a table column based on user role? (eg. only show a column for POWER user but hide it for BASIC user)
`

**Comment by mikesealey** [comment, marked as answer]
Hey @saz33m1 

This is possible in Budibase using conditions to update settings. 
https://docs.budibase.com/docs/conditions

Super quick demo:

I've got the sample data in a table component with only three columns displayed.
![image](https://github.com/user-attachments/assets/5571c353-8f07-44b6-8522-47bb00c7838e)

On the right I can enter the Conditions menu, and add a condition to Update Setting - Columns to show these columns if the Current User's Role is equal to the Power role.
![image](https://github.com/user-attachments/assets/b06fc754-056a-4918-a625-7c4278a03dce)

Similar principles can be used to update the appearance of specific cells or rows in the table component
![image](https://github.com/user-attachments/assets/27eb4486-aeb4-4f13-9334-72052600436d)

Here I can update the background colour of the row based on the employee level so that manager rows are displayed with a faint-red background colour.
![image](https://github.com/user-attachments/assets/35fdd033-bcd7-40ec-af62-de6ee7796a0d)

Read more about table conditional formatting here: https://docs.budibase.com/docs/table#conditional-formatting

You should also keep an eye out next week for v3 launches which may have more features relevant to your use-case. https://budibase.com/lp/3/
